Source/Core/InputCommon/InputProfile.cpp

```diff
@@ -0,0 +1,187 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Common/FileSearch.h"
+#include "Common/FileUtil.h"
+
+#include "Core/Config/WiimoteInputSettings.h"
+#include "Core/Core.h"
+#include "Core/HW/Wiimote.h"
+#include "Core/HotkeyManager.h"
+
+#include "InputCommon/ControllerInterface/ControllerInterface.h"
+#include "InputCommon/InputConfig.h"
+#include "InputCommon/InputProfile.h"
+
+#include <algorithm>
+#include <iterator>
+
+namespace
+{
+  enum class CycleDirection : int
+  {
+    Forward = 1,
+    Backward = -1
+  };
+
+  int s_wiimote_profile_index = 0;
+  const int s_display_message_ms = 3000;
+
+  std::vector<std::string> GetProfilesForDevice(InputConfig* device_configuration)
+  {
+    const std::string device_profile_root_location(File::GetUserPath(D_CONFIG_IDX) + "Profiles/" + device_configuration->GetProfileName());
+    return Common::DoFileSearch({ device_profile_root_location }, { ".ini" });
+  }
+
+  std::string GetProfile(CycleDirection cycle_direction, int& profile_index, const std::vector<std::string>& profiles)
+  {
+    // update the index and bind it to the number of available strings
+    auto positive_modulo = [](int& i, int n) {i = (i % n + n) % n;};
+    profile_index += static_cast<int>(cycle_direction);
+    positive_modulo(profile_index, static_cast<int>(profiles.size()));
+
+    return profiles[profile_index];
+  }
+
+  void UpdateProfile(const std::string& profile_filename, const std::vector<ControllerEmu::EmulatedController*> controllers)
+  {
+    std::string base;
+    SplitPath(profile_filename, nullptr, &base, nullptr);
+
+    IniFile ini_file;
+    if (ini_file.Load(profile_filename))
+    {
+      Core::DisplayMessage("Loading input profile: " + base, s_display_message_ms);
+
+      for (auto* controller : controllers)
+      {
+        controller->LoadConfig(ini_file.GetOrCreateSection("Profile"));
+        controller->UpdateReferences(g_controller_interface);
+      }
+    }
+    else
+    {
+      Core::DisplayMessage("Unable to load input profile: " + base, s_display_message_ms);
+    }
+  }
+
+  std::vector<ControllerEmu::EmulatedController*> GetControllersForDevice(InputConfig* device_configuration)
+  {
+    std::size_t size = device_configuration->GetControllerCount();
+
+    std::vector<ControllerEmu::EmulatedController*> result;
```

Comment:
This can just be:

```cpp
std::vector<ControllerEmu::EmulatedController*> result(size);
```
Given it's only a vector of pointers and default initializing those are trivial.